peekingduck/pipeline/nodes/model/hrnetv1/hrnet_files/detector.py

```diff
@@ -0,0 +1,158 @@
+
+        Args:
+            frame (np.ndarray): input image in numpy array
+            bboxes (np.ndarray): array of detected bboxes
+
+        Returns:
+            Tuple[np.ndarray, np.ndarray, Tuple[int, int]]: array of cropped bboxes, \
+            transformation matrices and original frame size
+        """
+        frame = frame / 255.0
+        frame_size = (frame.shape[1], frame.shape[0])
+        cropped_size = (self.resolution['width'], self.resolution['height'])
+
+        projected_bbox = project_bbox(bboxes, frame_size)
+        center_bbox = box2cs(projected_bbox, self.resolution['width'] / self.resolution['height'])
+        cropped_imgs, affine_matrices = crop_and_resize(frame, center_bbox, cropped_size)
+
+        return np.array(cropped_imgs), affine_matrices, frame_size
+
+    def postprocess(self, heatmaps: np.ndarray,  # pylint: disable=too-many-locals
+                    affine_matrices: np.ndarray,
+                    cropped_frames_scale: List[int],
+                    frame_size: Tuple[int, int]) -> Tuple[np.ndarray,
+                                                          np.ndarray,
+                                                          np.ndarray,
+                                                          np.ndarray]:
+        """Postprocessing function to process output heatmaps to required keypoint arays
+
+        Args:
+            heatmaps (np.ndarray): Output heatmaps from hrnet network
+            affine_matrices (np.ndarray): transformation matrices of preprocess cropping
+            cropped_frames_scale (List[int]): Shape of cropped bboxes
+            frame_size (Tuple[int, int]): Size of original image
+
+        Returns:
+            Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]: \
+            tuple containing array of bboxes and pose related info i.e coordinates,
+            scores, connections
+        """
+        batch, out_h, out_w, num_joints = heatmaps.shape
+        heatmaps_reshaped = reshape_heatmaps(heatmaps)
+
+        max_idxs = np.argmax(heatmaps_reshaped, 2)
+        kp_scores = np.amax(heatmaps_reshaped, 2)
+        keypoints = np.repeat(max_idxs, 2).reshape(batch, num_joints, -1).astype(np.float32)
+        keypoints[:, :, 0] = (keypoints[:, :, 0]) % out_w
+        keypoints[:, :, 1] = np.floor((keypoints[:, :, 1]) / out_w)
+
+        keypoints = scale_transform(keypoints,
+                                    in_scale=[out_w, out_h],
+                                    out_scale=cropped_frames_scale)
+        keypoints = affine_transform_xy(keypoints, affine_matrices)
+        keypoints, kp_masks = get_valid_keypoints(
+            keypoints, kp_scores, batch, self.min_score)
+
+        normalized_keypoints = keypoints / frame_size
+
+        keypoint_conns = get_keypoint_conns(normalized_keypoints, kp_masks)
+        keypoint_bboxes = get_bbox_of_poses(normalized_keypoints, kp_masks)
+        return normalized_keypoints, kp_scores, keypoint_conns, keypoint_bboxes
```

Comment:
Can remove `keypoint_bboxes` since we are not going to output bboxes

Reply:
okay, removed.